LLVM 20.1.4 ~python depends on python@3.8+

LLVM build depends on Python 3.8+ for build step only when variant ~python is used:
https://github.com/spack/spack/blob/e4f9e736716c7a2a22921e47b730d2a047a3e80d/var/spack/repos/spack_repo/builtin/packages/llvm/package.py#L416

However, spack itself is still happy with Python 3.6. This means that on an older system where the default Python is 3.6, one runs into this error:
```
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/cmake' '-G' 'Ninja' '-DCMAKE_INSTALL_PREFIX:STRING=/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/llvm-20.1.4-v6gptx6a6jvj2xfelyaxzibbiztlrlji' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/llvm-20.1.4-v6gptx6a6jvj2xfelyaxzibbiztlrlji/lib;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/llvm-20.1.4-v6gptx6a6jvj2xfelyaxzibbiztlrlji/lib64' '-DCMAKE_PREFIX_PATH:STRING=/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/binutils-2.44-dfdr4ygzphzaxnaun445fqoczjteteok;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/compiler-wrapper-1.0-tadynfg466zjtj7qulyhd5aaqol2suey;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/hwloc-2.11.1-imkkadjet4hf53pdv52ncvrbb3f74vwi;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/libedit-3.1-20240808-hxxvmjq2n7tgs7c24lcjh6lq3hrcystc;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/lua-5.3.6-tltqore44owqyeqtr3exj5w5gyzv7r4m;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/ninja-1.12.1-637wt5hufr5ojtoyn6mpwm5hrfofxeju;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/perl-data-dumper-2.173-ippwkxlazli6raagbwwp43j3ye3r5rrg;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/swig-4.1.1-slxna7txndfnrvlauidyz6rq7egwmgve;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/zstd-1.5.7-nowvdwa6lztawajlblbjnulwfo65anuy;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/libpciaccess-0.17-5z4imli7lrmxlqd3bzj5jftcy6bkoqje;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/libxml2-2.13.5-jcdyg4pjbhh5yzr7lzkmo3qe6kp7gipi;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/readline-8.2-hnwqgb6dpd5ndtyyqxeycrn5ydf4obml;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/pcre2-10.44-7rz4c5el7z2oc2ikjjiqcgn2h35aieqj;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/libiconv-1.18-2azxn6qgoyotehfdtdh64m6p2pooayml;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/xz-5.6.3-7g2a6akfehzelbnw5lgmv7waue7fakzj;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/zlib-ng-2.2.4-v2uxcn7vd7cwwcchh6derl4p7bljxvnp;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/ncurses-6.5-oqeubs3xntlykvwjtuozondddjyobxbz;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/gcc-runtime-8.5.0-fhkhg6quqewspnjz5jy42nsf2u632b5h' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DPYTHON_EXECUTABLE:STRING=/usr/bin/python3.6' '-DPython_EXECUTABLE:STRING=/usr/bin/python3.6' '-DPython3_EXECUTABLE:STRING=/usr/bin/python3.6' '-DLLVM_REQUIRES_RTTI:BOOL=ON' '-DLLVM_ENABLE_RTTI:BOOL=ON' '-DLLVM_ENABLE_LIBXML2:BOOL=OFF' '-DCLANG_DEFAULT_OPENMP_RUNTIME:STRING=libomp' '-DLIBOMP_USE_HWLOC:BOOL=ON' '-DLIBOMP_HWLOC_INSTALL_DIR:STRING=/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/hwloc-2.11.1-imkkadjet4hf53pdv52ncvrbb3f74vwi' '-DLLVM_ENABLE_ZSTD:BOOL=OFF' '-DCUDA_TOOLKIT_ROOT_DIR:STRING=IGNORE' '-DCUDA_SDK_ROOT_DIR:STRING=IGNORE' '-DCUDA_NVCC_EXECUTABLE:STRING=IGNORE' '-DLIBOMPTARGET_DEP_CUDA_DRIVER_LIBRARIES:STRING=IGNORE' '-DLIBOMPTARGET_ENABLE_DEBUG:BOOL=OFF' '-DLIBOMPTARGET_BUILD_AMDGPU_PLUGIN:BOOL=OFF' '-Dpython_executable:STRING=/usr/bin/python3.6' '-DLLDB_ENABLE_LIBEDIT:BOOL=ON' '-DLLDB_ENABLE_CURSES:BOOL=ON' '-DLLDB_ENABLE_LIBXML2:BOOL=ON' '-DLLDB_ENABLE_LUA:BOOL=ON' '-DLLDB_ENABLE_LZMA:BOOL=ON' '-DLLDB_CURSES_LIBS:STRING=/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/ncurses-6.5-oqeubs3xntlykvwjtuozondddjyobxbz/lib/libncurses.so;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/ncurses-6.5-oqeubs3xntlykvwjtuozondddjyobxbz/lib/libtinfo.so;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/ncurses-6.5-oqeubs3xntlykvwjtuozondddjyobxbz/lib/libncursesw.so;/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/ncurses-6.5-oqeubs3xntlykvwjtuozondddjyobxbz/lib/libtinfow.so' '-DLLDB_ENABLE_PYTHON:BOOL=OFF' '-DLLVM_BINUTILS_INCDIR:STRING=/gpfs/neptune/spack-stack/llvm-20.1.4/src/spack-4a22df5/opt/spack/linux-zen/binutils-2.44-dfdr4ygzphzaxnaun445fqoczjteteok/include' '-DOPENMP_ENABLE_LIBOMPTARGET:BOOL=OFF' '-DLLVM_ENABLE_Z3_SOLVER:BOOL=OFF' '-DLINK_POLLY_INTO_TOOLS:BOOL=ON' '-DBUILD_SHARED_LIBS:BOOL=OFF' '-DLLVM_BUILD_LLVM_DYLIB:BOOL=ON' '-DLLVM_LINK_LLVM_DYLIB:BOOL=OFF' '-DLLVM_USE_SPLIT_DWARF:BOOL=ON' '-DLIBCXX_ENABLE_STATIC_ABI_LIBRARY:BOOL=ON' '-DCMAKE_FIND_PACKAGE_PREFER_CONFIG:BOOL=ON' '-DCMAKE_FIND_USE_PACKAGE_ROOT_PATH:BOOL=OFF' '-DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_FIND_USE_SYSTEM_PATH:BOOL=OFF' '-DLLVM_TARGETS_TO_BUILD:STRING=X86' '' '-DLLVM_ENABLE_PROJECTS:STRING=lldb;clang;clang-tools-extra;flang;lld;polly' '-DLLVM_ENABLE_RUNTIMES:STRING=openmp;compiler-rt;libcxx;libcxxabi;libunwind' '-DRUNTIMES_CMAKE_ARGS:STRING=-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON;-DCMAKE_C_FLAGS:STRING=--gcc-install-dir=/usr/lib/gcc/x86_64-redhat-linux/8;-DCMAKE_CXX_FLAGS:STRING=--gcc-install-dir=/usr/lib/gcc/x86_64-redhat-linux/8' '/tmp/actions/spack-stage/spack-stage-llvm-20.1.4-v6gptx6a6jvj2xfelyaxzibbiztlrlji/spack-src/llvm'

1 error found in build log:
     29    -- lldb project is enabled
     30    -- mlir project is enabled
     31    -- openmp project is disabled
     32    -- polly project is enabled
     33    -- pstl project is disabled
     34    -- flang project is enabled
  >> 35    CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
     36      Could NOT find Python3: Found unsuitable version "3.6.8", but required is
     37      at least "3.8" (found /usr/bin/python3.6, found components: Interpreter)
     38    Call Stack (most recent call first):
     39      /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:592 (_FPHSA_FAILURE_MESSAGE)
     40      /usr/share/cmake/Modules/FindPython/Support.cmake:3165 (find_package_handle_standard_args)
     41      /usr/share/cmake/Modules/FindPython3.cmake:485 (include)

See build log for details:
  /tmp/actions/spack-stage/spack-stage-llvm-20.1.4-v6gptx6a6jvj2xfelyaxzibbiztlrlji/spack-build-out.txt
```

I assume the same minimum Python version 3.8 is required for the `+python` variant. It seems save to simply set the minimum requirement for Python to be 3.8?

--- a/repos/spack_repo/builtin/packages/llvm/package.py
+++ b/repos/spack_repo/builtin/packages/llvm/package.py
@@ -417,10 +417,12 @@ class Llvm(CMakePackage, CudaPackage, LlvmDetection, CompilerPackage):
         for runtime in ["libunwind", "libcxx", "compiler-rt"]:
             depends_on("cmake@:3.16", type="build", when="{0}=runtime".format(runtime))
         del runtime
+    depends_on("python@3.8:", when="@20: ~python", type="build")
     depends_on("python", when="~python", type="build")
     depends_on("pkgconfig", type="build")
 
     # Universal dependency
+    depends_on("python@3.8:", when="@20: +python")
     depends_on("python", when="+python")
 
     # clang and clang-tools dependencies

